contracts/signature/Eip712Signature.sol

```diff
@@ -0,0 +1,92 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+pragma solidity >=0.8.0 <0.9.0;
+
+/// @title EIP712Signature - A contract that extracts and inspects EIP-712 signatures appended to calldata.
+
+abstract contract EIP712Signature {
+    uint256 private nonce;
+
+    /**
+     * @dev Returns the current nonce value.
+     */
+    function eip712Nonce() public view returns (uint256) {
+        return nonce;
+    }
+
+    /**
+     * @dev Increments nonce.
+     */
+    function eip712BumpNonce() internal {
+        nonce = nonce + 1;
+    }
+
+    /**
+     * @dev When signature present in calldata, returns the address of the signer.
+     */
+    function eip712SignedBy() internal returns (address signer) {
+        if (msg.data.length >= 4 + 32 + 32 + 1) {
+            (
+                bytes calldata dataTrimmed,
```

Comment:
For the actual call full `msg.data` will be used, not only the `dataTrimmed`, am I right? Is it correct in general case, where such elongation of `msg.data` can interfere?

Reply:
Here we are operating under the assumption that the signature was produced using a a calldata input buffer, and then signature appended.

Thus the msg.data includes the "originalData" and the signature appended

To recreate the txModuleHash, we have to first remove the signature. Maybe dataTrimmed should be renamed originalData

Reply:
Same as commented before. Does the question still persist after reworking it?

Reply:
I originally meant the fact that for a real call signature bytes (from the last 65 bytes of the entire calldata) can be used (probably, partially) as part of the actual calldata. So, this leads to "signed `A` bytes as calldata, but `A || appendedSgianture` used in actual execution".